src/github.com/travis-ci/worker/backend/gce.go

```diff
@@ -644,6 +686,47 @@ func (p *gceProvider) imageForLanguage(language string) (*compute.Image, error)
 	return imagesByName[imageNames[len(imageNames)-1]], nil
 }
 
+func (p *gceProvider) imageForLanguage(language string) (*compute.Image, error) {
+	return p.imageByFilter(fmt.Sprintf(gceImageTravisCIPrefixFilter, language))
+}
+
+func (p *gceProvider) imageSelect(ctx gocontext.Context, startAttributes *StartAttributes) (*compute.Image, error) {
+	p.buildImageSelector()
```

Comment:
Could this cause a race condition if this method is called twice at the same time? There's a `if p.imageSelector != nil { return }` guard at the top of that method, but if the method is called twice very close to eachother they could both go past that?